middle_end/flambda2/to_cmm/to_cmm_primitive.ml

```diff
@@ -556,67 +575,116 @@ let unary_float_arith_primitive _env dbg width op arg =
   | Float32, Abs -> C.float32_abs ~dbg arg
   | Float32, Neg -> C.float32_neg ~dbg arg
 
+module Cmm_kind = struct
```

Comment:
A few remarks: we already have a plethora of similar types/modules to express some notion of cmm type/kind, so 1) this name if not descriptive enough to know what it means and why we need it, 2) this needs more documentation, 3) do we really need yet again another cmm type/kind/machtype/exttype/whatever ? 4) assuming we really need this new notion, this code does not have its place in this file: smart constructors functions for cmm expression should be in `cmm_helpers.ml` or in a separate file.

Note: we need to do a pass on all of the cmm type/kinds that we currently have to see how we could simplify things a bit, this is currently getting out of hand and confusing: we're currently stuck in an intermediate situation between the origin/vanilla cmm / upstream compiler where there tends to be few different types, vs the flambda2 way where we define as many types as we want so that we can be really specific, and I think that in conjunction with all of the subtelty of cmm, this is does not help.